src/Pangraph/GraphML/Parser.hs

```diff
@@ -1,28 +1,21 @@
-module Pangraph.GraphML.Parser(
-parseGraphToAlga,
-parseTemplateToPangraph,
-template,
-PT.Template
+module Pangraph.GraphML.Parser (
+parse,
+unsafeParse
 ) where
 
-import qualified Pangraph as P
-import qualified Text.XML.Hexml as H
-import qualified Data.ByteString as BS
-import qualified Pangraph.XMLTemplate as PT
-import qualified Algebra.Graph.HigherKinded.Class as H
+import qualified Data.ByteString            as BS
+import qualified Pangraph                   as P
+import qualified Pangraph.Internal.Hexml    as H
+import qualified Pangraph.XMLTemplate       as PT
 
+parse :: BS.ByteString -> Maybe P.Pangraph
+parse file =
```

Comment:
I believe you can simplify this function to:

`parse = fmap (PT.hexmlToPangraph PT.graphMLTemplate) . H.hexmlParse`.